lib/core/src/Cardano/Wallet/Primitive/BlockSummary.hs

```diff
@@ -123,17 +132,59 @@ toAscBlockEvents (ChainEvents bs) = Map.elems bs
 data BlockEvents = BlockEvents
     { slot :: !Slot
     , blockHeight :: !(Quantity "block" Word32)
-    , transactions :: [(Int, Tx)]
+    , transactions :: Sublist Tx
         -- ^ (Index of the transaction within the block, transaction data)
         -- INVARIANT: The list is ordered by ascending index.
-    , delegations :: [(Int, DelegationCertificate)]
+    , delegations :: Sublist DelegationCertificate
         -- ^ (Index of the delegation within the block, delegation data)
         -- INVARIANT: The list is ordered by ascending index.
     } deriving (Eq, Ord, Generic, Show)
 
+-- | A data type representing a sublist of a total list.
+-- Such a sublist typically arises by filtering and keeps
+-- track of the indices of the filtered list elements.
+--
+-- The main purpose of this data type is optimization:
+-- When processing whole 'Block', we want to avoid copying
+-- and redecorating the entire list of transactions in that 'Block';
+-- instead, we want to copy a pointer to this list.
```

Comment:
you mean the intent: when `All` then we process all content, then Some then only those pointed with ix (being map fst content), right?

Reply:
Yes, though the focus is on the consumer here: `fromEntireBlock` uses `All` and avoids redecorating. An external blockchain source will return `Some` with specific indices.